cheatsheets/Nodejs_security_cheat_sheet.md

```diff
@@ -0,0 +1,371 @@
+
+## Return sanitized user objects
+
+Information about the users of an application is among the most critical information about the application. Therefore, user information should be stored on a different database. User tables generally include fields like id, username, full name, email address, birth date, credit card number and in some cases social security numbers. Therefore, when querying and using user objects, you need to return only needed fields as it may be vulnerable to personal information disclosure. This is also correct for other objects stored on the database. If you just need a certain field of an object, you should never return it with all of its fields. As an example, you can use a function like the following whenever you need to get information on a user. By doing so, you can only return the fields that are needed for your specific operation. In other words, if you only need to list names of the users available, you are not returning their email addresses or credit card numbers in addition to their full names.
+
+```JavaScript
+exports.sanitizeUser = function(user) {
+  return {
+    id: user.id,
+    username: user.username,
+    fullName: user.fullName
+  };
+};
+```
+
+## Set request size limits
+
+Buffering and parsing of request bodies can be cumbersome for the server. If there is no limit on the size of requests, attackers can send request with large request bodies so that they can exhaust server memory or fill disk space. However, fixing a request size limit for all requests may not be the correct behavior, since some requests like those for uploading a file to the server have more content to carry on the request body. Also, input with a JSON type is more dangerous than a multipart input, since parsing JSON is a blocking operation. Therefore, you should set request size limits for different content types. You can accomplish this very easily with express middlewares as follows:
+
+```JavaScript
+app.use(express.urlencoded({ limit: “1kb” }));
+app.use(express.json({ limit: “1kb” }));
+app.use(express.multipart({ limit:”10mb” }));
+app.use(express.limit(“5kb”)); // this will be valid for every other content type
+```
+
+## Use strict mode
+
+JavaScript has some unsafe features that lies within the language itself. In order to remove these features, ES5 included a strict mode for developers. With this mode, errors that were silent previously are thrown. It also restricted optimizations for JavaScript engines. With strict mode, previously accepted bad syntax causes real errors. use "use strict". Because of these improvements, you should always use strict mode in your application. In order to enable strict mode, you just need to write _‘use strict’;_ on top of your code.
```

Comment:
Replace first sentence with something like `JavaScript has a number of unsafe and dangerous legacy features that should not be used.`

`use 'use strict'` isn't really a complete sentence

Put it into a JavaScript code block like the other examples?

Reply:
modified accordingly and added an example